assets/js/apps/search/components/ResultSection.jsx

```diff
@@ -0,0 +1,285 @@
+    onSelect: PropTypes.func.isRequired,
+    isSelected: PropTypes.bool.isRequired
+}
+
+export function PureResultList({ results, selectedItem, onItemSelect, error, isLoading }) {
+    let body;
+    const handleItemSelected = (id) => {
+        onItemSelect(id);
+    }
+
+    if (error) {
+        return (
+            // If there was an error during the search
+            <div className="result-section--msg result-section--error-msg">
+                <p>An error occurred. Please try another query, or refresh the page and try searching again.</p>
+            </div>
+        );
+    }
+
+    else if (isLoading) {
+        body = (
+            // If the search is in progress.
+            <tr>
+                <td colSpan="3">
+                    <div className="result-section--msg">
+                        <p>Loading...</p>
+                    </div>
+                </td>
+            </tr>
+        );
+    }
+
+    else if (!results ||
+        (Array.isArray(results) && results.length == 0)) {
+        // If the results are empty...
+        body = (
+            <tr>
+                <td colSpan="3">
+                    <div className="result-section--msg">
+                        <p>No results. Please adjust your search and try again.</p>
+                    </div>
+                </td>
+            </tr>
+        )
+    }
+
+    else {
+        // Render the results in table.
+        body = results.map((result) => (
+            <ResultRow key={result.id}
+                onSelect={handleItemSelected.bind(this, result.id)}
+                result={result}
+                isSelected={result.id === selectedItem}
+            />
+        ));
+    }
+
+    return (
+        <Table className="result-section__container" responsive hover>
+            <thead>
+                <tr>
+                    <th></th>
+                    <th>Name</th>
+                    <th>Size</th>
+                </tr>
+            </thead>
+            <tbody>
+                {body}
+            </tbody>
+        </Table>
+    )
+}
+
+PureResultList.propTypes = {
+    results: PropTypes.arrayOf(Object),
+    error: PropTypes.string,
+    isLoading: PropTypes.bool,
+    selectedItem: PropTypes.number,
+    onItemSelect: PropTypes.func
+}
+
+export function PureResultSection({ resultSets, selectedType,
+    onSelectType, selectedResult, onSelectResult, isLoading, error }) {
+    let counts;
+    if (!resultSets) {
+        resultSets = {};
+        counts = {
+            project: null,
+            experiment: null,
+            dataset: null,
+            datafile: null
+        }
+    } else {
+        counts = {};
+        for (let key in resultSets) {
+            counts[key] = resultSets[key].length;
+        }
+    }
+
+        let selectedEntry = getSelectedEntry(resultSets, selectedResult, selectedType);
+
+        const currentResultSet = resultSets[selectedType],
+            currentCount = counts[selectedType];
+        return (
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/gh/mytardis/mytardis/pullRequest?prid=7933948)
```suggestion
    return (
```